icu4c/source/test/fuzzer/collator_compare_fuzzer.cpp

```diff
@@ -10,23 +11,36 @@
 
 IcuEnvironment* env = new IcuEnvironment();
 
+static const std::array<icu_74::Collator::ECollationStrength, 5> kStrength = {
+    icu::Collator::PRIMARY,
+    icu::Collator::SECONDARY,
+    icu::Collator::TERTIARY,
+    icu::Collator::QUATERNARY,
+    icu::Collator::IDENTICAL
+};
+
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
   UErrorCode status = U_ZERO_ERROR;
 
   if (size < 2)
     return 0;
 
+  uint16_t rnd = *(reinterpret_cast<const uint16_t *>(data));
+
   std::unique_ptr<char16_t> compbuff1(new char16_t[size/4]);
   std::memcpy(compbuff1.get(), data, (size/4)*2);
   data = data + size/2;
```

Comment:
Managing data and size would be simpler if you used an icu::StringPiece. Initialize it early from the inputs. You update both pointer and length with remove_prefix(n) and then get the new values via data() and length(). (In C++17 we could use a std::string_view.)

Reply:
That is in the pre-existing code and I didn't change it.

Reply:
The existing code at least looked correct, using separate parts of the fuzzer input for the various pieces. The new code introduces an overlap between pieces, and fixing that will complicate tracking data+size. StringPiece will help.